docs/source/reference/storage.rst

```diff
@@ -204,100 +134,142 @@ and storage mounting:
     aws/gsutil cli) will see the latest data.
 
 .. note::
-    :code:`MOUNT` mode does not support the full POSIX interface and some file
-    operations may fail. Most notably, random writes and append operations are
-    not supported.
+    SkyPilot does not guarantee preservation of file permissions when attaching
+    object stores. You may need to set file permissions during task execution.
 
 .. note::
-    Storage only supports uploading directories (i.e., :code:`source` cannot be a file).
-    To upload a single file to a bucket, please put in a directory and specify the directory as the source.
-    To directly copy a file to a VM, please use regular :ref:`file mounts <file-mounts-example>`.
+    Symbolic links are handled differently in :code:`file_mounts` depending on whether object stores are used.
+    For object store mounts, symbolic links are not copied to remote.
+    For local ``file_mounts`` that are directly rsynced to the VM,
+    the symbolic links are directly copied, not their target data.
+    The targets must be separately mounted or else the symlinks may break.
 
-Creating a shared file system
------------------------------
+Common patterns
+---------------
 
-SkyPilot Storage can also be used to create a shared file-system backed by a remote object store (e.g., S3)
-that multiple tasks on different nodes can read and write to. This allows developers to pass files
-between workers and even use files as a medium for inter-process communication (IPC).
+Storing and sharing outputs of tasks
```

Comment:
Should we mention the mounting an existing bucket as well in the common pattern as well?

Reply:
Hmm that's covered in `Usage` at the top. Should we repeat it here?

Reply:
Ahh, I see. It might be good to avoid the duplication.

Reply:
+1, I think it's good to add a "Reading datasets" (from an existing bucket) here. Most orgs have such setups.